jsonschema-core.xml

```diff
@@ -953,35 +953,234 @@
+                        </t>
+                        <t>
+                            Note that if the schema to which "$recursiveRef" referrs does not
+                            contain "$recursiveAnchor" set to true, or if there are no other
+                            "$recursiveAnchor" keywords set to true anywhere further back in
+                            the dynamic scope, then "$recursiveRef"'s behavior is identical
+                            to that of "$ref".
+                        </t>
+                        <figure>
+                            <preamble>
+                                With this in mind, we can rewrite the previous example:
+                            </preamble>
+                            <artwork>
+<![CDATA[
+{
+    "$schema": "http://json-schema.org/draft-08/schema#",
+    "$id": "https://example.com/basic",
+    "$recursiveAnchor": true,
+
+    "properties": {
+        "name": {
+            "type": "string"
+        },
+        "recursive": {
+            "$recursiveRef": "#"
+        }
+    }
+}
+
+{
+    "$schema": "http://json-schema.org/draft-08/schema#",
+    "$id": "https://example.com/extension",
+    "$recursiveAnchor": true,
+
+    "$ref": "basic",
+    "properties": {
+        "things": {
+            "type": "array"
+            "items": {
+                "$recursiveRef": "#"
+            }
+        }
+    }
+}
+]]>
+                            </artwork>
+                            <postamble>
+                                Now lets consider the evaluation of the "extension" schema.
+                                Note that the "$ref": "basic" was not changed, as it works
+                                just fine as a normals static reference.  And the
+                                "$recursiveRef" in the "extended" schema does not behave at
+                                all differently, because the "$recursiveAnchor" in its
+                                referred schema is the outermost "$recursiveAnchor" in the
+                                dynamic scope.  However, the "$recursiveRef" in the "basic"
+                                schema referrs to a "$recursiveAnchor" that is not the
+                                outermost such keyword in the dynamic scope.  That is still
+                                the "$recursiveAnchor" in the "extension" schema.
+                                Therefore, when processing starts with the extension
+                                schema, the "$recursiveRef" in the basic schema actually
+                                referrs to the "extension" schema's root schema.
+                            </postamble>
+                        </figure>
+                    </section>
+                </section>
+
+                <section title="Guarding Against Inifinite Recursion">
+                    <t>
+                        A schema MUST NOT be run into an infinite loop against an instance. For
```

Comment:
> A schema MUST NOT be run into an infinite loop against an instance.

This doesn't really make sense.  The explanation afterward does, but I don't get this sentence.

Reply:
I believe I just copy-pasted that from wherever it was before, so if we want to consider reworking it please file a new issue on it.  I'm fine with discussing it, it's just not changed here so I don't want to add to the PR.